packages/runtime-common/realm.ts

```diff
@@ -1783,6 +1789,31 @@ export class Realm {
     };
   }
 
+  private async visibility(): Promise<RealmVisibility> {
+    if (!this.visibilityCache) {
```

Comment:
it's completely a style chioce, but I would have first tried to return the cached value, and then if there is no cached value do the work to calculate the cached value. that way there is one less `{}`. Also, if you wanted to actually cache the promise for the visbility, its a lot easier. (which way if there are a bunch of calls to realm info at the same time, they can all share work instead of having to recalculate visibility). I don't think its really necessary to cache the promise, but its a nice touch if you wanted to do that.